extension/json/json_functions/json_pretty.cpp

```diff
@@ -0,0 +1,32 @@
+#include "json_executors.hpp"
+
+namespace duckdb {
+
+//! Pretty Print a given JSON Document
+string_t PrettyPrint(yyjson_val *val, yyjson_alc *alc, Vector &result) {
+	D_ASSERT(alc);
+	idx_t len;
+	auto data =
+	    yyjson_val_write_opts(val, JSONCommon::WRITE_PRETTY_FLAG, alc, reinterpret_cast<size_t *>(&len), nullptr);
+	return StringVector::AddString(result, data, len);
+}
+
+static void PrettyPrintFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	auto json_type = args.data[0].GetType();
+	D_ASSERT(json_type == LogicalType::VARCHAR || json_type == LogicalType::JSON());
+
+	JSONExecutors::UnaryExecute<string_t>(args, state, result, PrettyPrint);
```

Comment:
Should we pretty-print nulls?
I believe the UnaryExecutor behaves as null in -> null out, not calling the function.

I guess nulls are represented differently in json, so maybe this is not a problem?

Reply:
`NULL`s in JSON are different from SQL nulls.

As far as I can see the following behaviour should be the expected one:
```sql
D .mode duckbox
D SELECT json_pretty(null);
┌───────────────────┐
│ json_pretty(NULL) │
│      varchar      │
├───────────────────┤
│                   │
└───────────────────┘
D SELECT json_pretty('null'::json);
┌─────────────────────────────────────┐
│ json_pretty(CAST('null' AS "json")) │
│               varchar               │
├─────────────────────────────────────┤
│ null                                │
└─────────────────────────────────────┘
D .mode ascii
D SELECT json_pretty('[1,2,null, {"a": null},  42]'::json);
json_pretty(CAST('[1,2,null, {"a": null},  42]' AS "json"))[
    1,
    2,
    null,
    {
        "a": null
    },
    42
]D 
```

I would include them all in the unit test.